File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        const n = numbers.length;
        for(let i = 0; i < n - 1; i++) {
            for(let j = i + 1; j < n; j++) {
                if(numbers[i] + numbers[j] === target) {
                    return [i + 1, j + 1];
                }
            }
        }
        return null;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> tmp = nums;
        sort(tmp.begin(), tmp.end());
        int n = tmp.size();
        int i = 0, j = n - 1;

        while(tmp[i] + tmp[j] != target){
            int sum = tmp[i] + tmp[j];
            if(sum > target) j--;
            else if(sum < target) i++;
        }

        int num1 = tmp[i];
        int num2 = tmp[j];

        vector<int> ans;
        for(int i = 0; i < n; i++){
            if(nums[i] == num1 || nums[i] == num2){
                ans.push_back(i);
                if(ans.size() == 2) break;
            }
        }

        return ans;
    }
};